### ISSUE TITLE
implement text splitter

### ISSUE THREAD

**Issue by pelikhan** [microsoft/genaiscript#435]
- text
- markdown
- code
- pdf
- ...

**Comment by cmaneu** [comment]
@pelikhan Just to be sure, is that issue is about automagically splitting the input contents, so we can run the same prompt over a large file, even if the file exceeds the model context token window?

cc @sinedied

**Comment by pelikhan** [comment]
yes the issue is about there to cut/chunk large text files to rag or split into multiple prompts.

It's a bit outdated, there is encoders.chunk now that mostly does that.
